lte/gateway/python/magma/subscriberdb/tests/client_tests.py

```diff
@@ -272,3 +280,38 @@ def test_detach_deleted_subscribers(self, s6a_service_mock_stub):
         mock.DeleteSubscriber.future.assert_called_once_with(
             DeleteSubscriberRequest(imsi_list=["101", "303"]),
         )
+
+    @ unittest.mock.patch(
+        'magma.common.service_registry.ServiceRegistry.get_rpc_channel',
+    )
+    def test_get_subscribers_no_updates(self, get_grpc_mock):
```

Comment:
Does this also test how things work with multiple pages?

Reply:
This test suite has a mock `ListSubscribers` for the test cases to work with, and the mock implements the same logic as on the cloud, as in it does return multiple pages of subscribers on request, but applies the flat digest logic on the first page (as in if the digests match for the request for the first page, it returns `no_updates=True`).

This helps test the logic for the agw, where if it receives a `no_updates=True` for any page, it stops loading the succeeding pages.